server.py

```diff
@@ -70,14 +73,27 @@ def post_collect():
             message="Missing 'x-rh-identity' header"
         ), 401
 
+    identity = json.loads(base64.b64decode(b64_identity))
+    account_id = identity.get('identity', {}).get('account_number')
+    APP.logger.debug('Account_id: %s', account_id)
+
+    # Check if this account has been processed within the window
+    if account_id and collector.utils.processed(account_id):
```

Comment:
I think this should be:
```suggestion
    if not account_id or collector.utils.processed(account_id):
```
Because in case `account_id` is `None`, this would pass and a worker would get started. I don't think we want that.

Reply:
That's my intention to execute it, at least from `outlier detection` use case perspective.  The current contract we have from Host Inventory Service is that a valid `x-rh-header`(even without `account_id` in it) would get back hosts info from Host Inventory Service.

Reply:
Oh, I see.. let's keep it this way then.